File: app/src/main/java/im/kirillt/yandexmoneyclient/events/payment/PaymentProcessEvent.java
package im.kirillt.yandexmoneyclient.events.payment;

import com.yandex.money.api.methods.ProcessPayment;

import java.io.IOException;

import de.greenrobot.event.EventBus;
import im.kirillt.yandexmoneyclient.YMCApplication;
import im.kirillt.yandexmoneyclient.events.AnyErrorEvent;
import im.kirillt.yandexmoneyclient.utils.ResponseReady;

public class PaymentProcessEvent {
    final String requestId;

    public PaymentProcessEvent(String requestId) {
        this.requestId = requestId;
    }

    public void process() {
        try {
            YMCApplication.auth2Session.enqueue(new ProcessPayment.Request(requestId), new ResponseReady<ProcessPayment>() {
                @Override
                protected void failure(Exception exception) {
                    EventBus.getDefault().post(new AnyErrorEvent(exception));
                }

                @Override
                protected void response(ProcessPayment response) {
                    EventBus.getDefault().post(new PaymentProcessResultEvent(response));
                }
            });
        } catch (IOException e) {
            e.printStackTrace();
            EventBus.getDefault().post(new AnyErrorEvent(e));
        }
    }
}

File: app/src/main/java/im/kirillt/yandexmoneyclient/events/payment/PaymentProcessEvent.kt
package im.kirillt.yandexmoneyclient.events.payment

import com.yandex.money.api.methods.ProcessPayment

import de.greenrobot.event.EventBus
import im.kirillt.yandexmoneyclient.YMCApplication
import im.kirillt.yandexmoneyclient.events.AnyErrorEvent

class PaymentProcessEvent(internal val requestId: String) {

    fun process() {
        try {
            val response = YMCApplication.client.execute(ProcessPayment.Request(requestId))
            EventBus.getDefault().post(PaymentProcessResultEvent(response));
        } catch (e: Exception) {
            e.printStackTrace()
            EventBus.getDefault().post(AnyErrorEvent(e))
        }

    }
}
